### PR TITLE
primary info form + refactor

### PR DESCRIPTION
**Pull request by jjwon0**
- moved `CheckIn` related components to the route
- decoupled `CheckInForm` from `CheckIn` and gave it a separate container
- added some state for `CheckIn` (my implementation may be suboptimal; I'm just using `state.check-in` as a dumping ground for different form fields, but we may want to store some other state on this later, in which we should probably move this to like `state.check-in.fields` or something like that.
- added all the primary info form fields

**TODO**: add api endpoints on the server to receive this data (when it's complete), no clue how to like organize koa server stuff though.

### CODE REVIEW

**Comment by altonzheng** [comment]
tests? jk LOL

File: src/routes/CheckIn/modules/check-in.js [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
+    const fields = action.payload
+    const newState = {}
+
+    for (let field in fields) {
+      if (fields[field].touched && fields[field].value) {
+        newState[field] = fields[field].value
+      }
+    }
+
+    return Object.assign({}, state, newState)
```
Comment (altonzheng) [inline src/routes/CheckIn/modules/check-in.js]: Does ES6 syntax work here?

ie.
`return {..state, newState}`

File: src/routes/CheckIn/components/CheckInForm.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        <input type="checkbox" {...ethnicity} />Caucasian
+        <input type="checkbox" {...ethnicity} />Latino
+        <input type="checkbox" {...ethnicity} />Native American
+        Other <input type="text" {...ethnicity} />
+        {ethnicity.touched && ethnicity.error && <div>{ethnicity.error}</div>}
+      </div>
+
+      <div>
+        <label>Primary Language</label>
+        <input type="checkbox" {...language} />English
```
Comment (altonzheng) [inline src/routes/CheckIn/components/CheckInForm.js]: What is this doing? `{...language} `

Not sure what the reason is, but when you select a checkbox, all of them are selected.

File: src/routes/CheckIn/modules/check-in.js (hunk 1 above)
Comment (jjwon0) [inline src/routes/CheckIn/modules/check-in.js]: It works but it's preferable to keep state immutable, so we should create new objects instead.

File: src/routes/CheckIn/components/CheckInForm.js (hunk 2 above)
Comment (jjwon0) [inline src/routes/CheckIn/components/CheckInForm.js]: Yeah I screwed this up 😁, made it cleaner hopefully.

Still something that looks a little weird cause the Other field gets populated, will just make that an issue and address it later.

**Comment by jjwon0** [comment]
No clue how to write React tests LOOOOL

File: src/routes/CheckIn/components/CheckInForm.js (hunk 2 above)
Comment (jjwon0) [inline src/routes/CheckIn/components/CheckInForm.js]: `{...language}` destructures the values of `language` into the `input` field.  In this case, `languages` is actually an object (from `redux-form`) which has more than just the value itself -- it has methods like `onChange/onBlur` and attributes like `valid/dirty/value` which help it maintain its own state.

**Comment by jjwon0** [comment]
rebased

File: src/routes/CheckIn/containers/CheckInFormContainer.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  'hasServedInTheMilitary',
+
+  'primaryHealthcareLocation',
+
+  'isHomeless',
+  'lengthOfHomelessness',
+]
+
+const validate = (values) => {
+  const errors = {}
```
Comment (jjwon0) [inline src/routes/CheckIn/containers/CheckInFormContainer.js]: will fix all of these later when we add some sort of nice validation library

**Pull request merged**